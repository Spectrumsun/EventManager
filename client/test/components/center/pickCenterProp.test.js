import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { createMemoryHistory } from 'history';
import Display from '../../../src/components/Center/pickCenterProp';

const props = {
  selectCenter: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  error: false,
  history: createMemoryHistory(),
  center: [
    {
      id: 29,
      centerName: 'farm center',
      city: 'Yaba',
      address: 'No 33, yaba road lagos',
      availability: 'yes',
      imageurl: 'llal',
      imageId: 'lol',
      about: 'lalalla',
      facility: ['car', 'fan', 'video']
    },
    {
      id: 9,
      centerName: 'farm center',
      city: 'Yaba',
      address: 'No 33, yaba road lagos',
      availability: 'yes',
      imageurl: 'llal',
      imageId: 'lol',
      about: 'lalalla',
      facility: ['car', 'fan', 'video']
    }
  ],
  page: 2,
  facility: ['car', 'fan', 'video'],
  id: 9,
  centerName: 'farm center',
  address: 'No 33, yaba road lagos',
  image: 'llal'
};

const shallowWrapper = shallow(<Display {...props} />);

describe('<Display />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Display {...props} />);
  });

  it('should render the HomePage', () => {
    shallow(<Display {...props} />);
  });

  it('should render initial layout of Display', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  
  it('should have image on layout', () => {
    expect(wrapper.find('img').length).toEqual(1);
  });

  it('should call clicked method when sending the event id to the event state', () => {
    sinon.spy(shallowWrapper.instance(), 'clicked');
    shallowWrapper.instance().clicked();
    expect(shallowWrapper.instance().clicked.calledOnce).toEqual(true);
  });
});
