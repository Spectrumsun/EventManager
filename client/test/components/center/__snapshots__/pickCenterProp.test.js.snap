// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> should render initial layout of Display 1`] = `
Array [
  <div
    className="card"
    style={
      Object {
        "width": "12rem",
      }
    }
  >
    <img
      alt="Card cap"
      className="card-img-top"
      src="llal"
      style={
        Object {
          "height": "10",
          "width": "20",
        }
      }
    />
    <div
      className="theHeight"
    >
      <div
        className="card-body text-capitalize"
      >
        <h6>
          <Link
            replace={false}
            style={
              Object {
                "color": "#2E3A41",
              }
            }
            target="_blank"
            to="/centers/9"
          >
            <strong>
              farm center
            </strong>
          </Link>
        </h6>
        <h6>
          No 33, yaba road lagos
        </h6>
        <div
          className="scrollbar"
        >
          <h6>
            <strong>
              Facility
            </strong>
          </h6>
          <ul
            className="list-group text-capitalize"
          >
            <li>
              car
            </li>
            <li>
              fan
            </li>
            <li>
              video
            </li>
          </ul>
        </div>
        <br />
      </div>
    </div>
    <div
      className="buttingFix"
    >
      <button
        className="btn btn-primary btn-sm"
        data-dismiss="modal"
        onClick={[Function]}
        type="button"
      >
        Select
      </button>
    </div>
  </div>,
]
`;
